invokeai/backend/tiles/tiles.py

```diff
@@ -63,31 +97,122 @@ def calc_tiles_with_overlap(
+            bottom = image_height
+
         for tile_idx_x in range(num_tiles_x):
-            cur_tile = get_tile_or_none(tile_idx_y, tile_idx_x)
-            top_neighbor_tile = get_tile_or_none(tile_idx_y - 1, tile_idx_x)
-            left_neighbor_tile = get_tile_or_none(tile_idx_y, tile_idx_x - 1)
+            # Calculate the left & right coordinate of each tile
+            left = tile_idx_x * (tile_size_x - overlap_x)
+            right = min(left + tile_size_x, image_width)
+            # For the last tile in the row adjust right to be the width of the image
+            if tile_idx_x == num_tiles_x - 1:
+                right = image_width
 
-            assert cur_tile is not None
+            tile = Tile(
+                coords=TBLR(top=top, bottom=bottom, left=left, right=right),
+                overlap=TBLR(top=0, bottom=0, left=0, right=0),
+            )
 
-            # Update cur_tile top-overlap and corresponding top-neighbor bottom-overlap.
-            if top_neighbor_tile is not None:
-                cur_tile.overlap.top = max(0, top_neighbor_tile.coords.bottom - cur_tile.coords.top)
-                top_neighbor_tile.overlap.bottom = cur_tile.overlap.top
+            tiles.append(tile)
 
-            # Update cur_tile left-overlap and corresponding left-neighbor right-overlap.
-            if left_neighbor_tile is not None:
-                cur_tile.overlap.left = max(0, left_neighbor_tile.coords.right - cur_tile.coords.left)
-                left_neighbor_tile.overlap.right = cur_tile.overlap.left
+    return calc_overlap(tiles, num_tiles_x, num_tiles_y)
 
-    return tiles
+
+def calc_tiles_min_overlap(
+    image_height: int,
+    image_width: int,
+    tile_height: int,
+    tile_width: int,
+    min_overlap: int = 0,
+    round_to_8: bool = False,
+) -> list[Tile]:
+    """Calculate the tile coordinates for a given image shape under a simple tiling scheme with overlaps.
+
+    Args:
+        image_height (int): The image height in px.
+        image_width (int): The image width in px.
+        tile_height (int): The tile height in px. All tiles will have this height.
+        tile_width (int): The tile width in px. All tiles will have this width.
+        min_overlap (int): The target minimum overlap between adjacent tiles. If the tiles do not evenly cover the image
+            shape, then the overlap will be spread between the tiles.
+
+    Returns:
+        list[Tile]: A list of tiles that cover the image shape. Ordered from left-to-right, top-to-bottom.
+    """
+    assert image_height >= tile_height
+    assert image_width >= tile_width
+    assert min_overlap < tile_height
+    assert min_overlap < tile_width
+
+    num_tiles_x = math.ceil((image_width - min_overlap) / (tile_width - min_overlap)) if tile_width < image_width else 1
+    num_tiles_y = (
+        math.ceil((image_height - min_overlap) / (tile_height - min_overlap)) if tile_height < image_height else 1
+    )
```

Comment:
This is @JPPhoto code so hopefully, he can comment on it.

Reply:
@RyanJDick The original code did not have asserts and had the `else 1` conditions because you could request a tile size larger than the image size and the code would be resilient. I agree that with the asserts your change is valid, but I personally favor the former approach. It's your call.

Reply:
Got it. So, options are:
1. Allow tile dimensions larger than the image and just clip to the image dimensions.
2. Raise if tile dimensions exceed the image dimensions.

I have no strong preference. Happy with either.

Reply:
Discussed with @JPPhoto and the asserts on tiles size > image size have been removed and we are staying with `else 1`